mllib/src/main/scala/org/apache/spark/ml/feature/IDF.scala

```diff
@@ -175,13 +185,25 @@ object IDFModel extends MLReadable[IDFModel] {
     private val className = classOf[IDFModel].getName
 
     override def load(path: String): IDFModel = {
+      import org.json4s.DefaultFormats
+      implicit val format = DefaultFormats
       val metadata = DefaultParamsReader.loadMetadata(path, sc, className)
+
       val dataPath = new Path(path, "data").toString
       val data = sparkSession.read.parquet(dataPath)
-      val Row(idf: Vector) = MLUtils.convertVectorColumnsToML(data, "idf")
-        .select("idf")
-        .head()
-      val model = new IDFModel(metadata.uid, new feature.IDFModel(OldVectors.fromML(idf)))
+      val model = if (majorVersion(metadata.sparkVersion) >= 3) {
```

Comment:
Yeah some classes use the Spark version, some use a versioning mechanism. Let me ask dev@ whether one is preferred. For example I'm not sure if maybe it's a fine practice to just select all columns and see what is available rather than switch on Spark version.